presto-spi/src/main/java/io/prestosql/spi/connector/ColumnMetadata.java

```diff
@@ -156,4 +181,82 @@ public boolean equals(Object obj)
                 Objects.equals(this.extraInfo, other.extraInfo) &&
                 Objects.equals(this.hidden, other.hidden);
     }
+
+    public static Builder builder()
```

Comment:
I would like to pass required parameters here (name and type).

Reply:
it' breaks code layout and thus makes it harder to read

Reply:
But it less error prone.

Reply:
i cannot do this.

Reply:
@kokosing  I feel strongly about it.

- builder pattern is a builder pattern no matter how many properties are required and how many are optional
- this simply reads better if every property is set the same way
- i can have Jackson use the builder for deserialization, allowing me to use private constructor in my value class
- it's not error-prone, as required properties are validated in the ctor of the value class

Reply:
> it's not error-prone

Validation is at runtime vs compilation time.

> this simply reads better if every property is set the same way

This is a matter of taste.

Up to you.